test/irqbalance.bats

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env bats
+# vim: set syntax=sh:
+
+load helpers
+
+function detect_irqbalance_config() {
+	# debian/ubuntu
+	[ -f /etc/default/irqbalance ] && echo "/etc/default/irqbalance"
+	# fedora/centos/RHEL
+	[ -f /etc/sysconfig/irqbalance ] && echo "/etc/sysconfig/irqbalance"
+	# default
+	echo ""
+}
+
+function setup_file() {
+	if ! command -v irqbalance > /dev/null; then
+		skip "irqbalance not found."
+	fi
+	IRQBALANCE_CONF=$(detect_irqbalance_config)
+	if [ -z "$IRQBALANCE_CONF" ]; then
+		echo "error: unable to find irqbalance config file"
+		return 1
+	fi
+	CONFIGLET="$CRIO_CONFIG_DIR/99-irqbalance.conf"
+
+	mkdir -p "/etc/sysconfig"
+	BANNEDCPUS_CONF="/etc/sysconfig/orig_irq_banned_cpus"
+
+	export IRQBALANCE_CONF
+	export CONFIGLET
+	export BANNEDCPUS_CONF
+}
+
+function setup() {
+	setup_test
+	stop_crio
+	# we don't unconditionally restore because the irqbalance package may be missing
+	# from the test system. if this is the case, one less thing to worry about, we can't
+	# pollute the system state anyway :)
+	if [ -f "$IRQBALANCE_CONF" ]; then
+		cp -v "$IRQBALANCE_CONF" "$IRQBALANCE_CONF".bkp
+	else
+		# placeholder to make the rest of the suite simpler.
+		touch "$IRQBALANCE_CONF"
+		touch /tmp/.test_owns_irqbalance_conf
+	fi
+	if [ -f "$BANNEDCPUS_CONF" ]; then
+		cp -v "$BANNEDCPUS_CONF" "$BANNEDCPUS_CONF".bkp
+	else
+		# empty file is fine and meaningful
+		touch "$BANNEDCPUS_CONF"
+		touch /tmp/.test_owns_bannedcpus_conf
+	fi
+}
+
+function teardown() {
+	cleanup_test
+	stop_crio
+	# see setup about why we have these conditionals
+	if [ -f "$IRQBALANCE_CONF".bkp ]; then
+		mv -v "$IRQBALANCE_CONF".bkp "$IRQBALANCE_CONF"
+	elif [ -f /tmp/.test_owns_irqbalance_conf ]; then
+		rm -f "$IRQBALANCE_CONF"
+		rm -f /tmp/.test_owns_irqbalance_conf
+	fi
+
+	if [ -f "$BANNEDCPUS_CONF".bkp ]; then
+		mv -v "$BANNEDCPUS_CONF".bkp "$BANNEDCPUS_CONF"
+	elif [ -f /tmp/.test_owns_bannedcpus_conf ]; then
+		rm -f "$BANNEDCPUS_CONF"
+		rm -f /tmp/.test_owns_bannedcpus_conf
+	fi
+}
+
+@test "irqbalance cpu ban list save" {
+	# given
+	if ! grep -Eq '^[1,3,7,f]{1,}$' /proc/irq/default_smp_affinity; then
```

Comment:
I assume that what we wanna check here is that all cpus  are permitted for IRQs by default, so that the binary representation is all "1"s.  If the number of CPUs is not a multiple of 4 there will be other hex digits than "f" in the file, so we need to check for them also.
ej: 2 CPUs (as in the github ci ) -> "3"

Reply:
right (and unexpected). This deserves a comment in the test code.

Reply:
Did we add this comment?

Reply:
still relevant